### DISCUSSION TITLE
Change sampler using ask/tell interface

### DISCUSSION THREAD

**Discussion by thomasahle** [optuna/optuna#5614] [Q&A]
I have a function f(args) -> 0 or 1, that is highly noisy.
Optuna lets me try different args and find the ones that maximize E[f(args)].

However, `study.ask()` need to balance exploration and exploitation.
A the end of my ask/tell loop, I would like Optuna to give the me args that maximize E[f(args)].
This is not the same as `study.best_trial`, which just return a random trial where `f(args)` was 1.

The question is whether I can "switch sampler" at the last `study.ask` to get one that's 100% exploitation?

**Comment by nzw0301** [comment]
At least we can switch the sampler as shown in https://github.com/optuna/optuna/issues/2635#issuecomment-828886427. 

> 100% exploitation

I think this depends on the sampler's algorithm.
